context/src/main/java/io/opentelemetry/context/LazyStorage.java

```diff
@@ -48,17 +48,11 @@ private static ContextStorage createStorage(AtomicReference<Throwable> deferredS
       return DefaultContext.threadLocalStorage();
     }
 
-    if (providers.size() == 1) {
-      ContextStorageProvider provider = providers.get(0);
-      try {
-        return provider.get();
-      } catch (Throwable t) {
-        deferredStorageFailure.set(t);
-        return DefaultContext.threadLocalStorage();
+    if (providerClassName.isEmpty()) {
```

Comment:
@dengliming Sorry if my suggestions weren't clear, thanks for bearing with me. We have these success cases

- 1 provider, providerClassName empty
- 1 provider, providerClassName matches the provider <- currently missing
- multiple providers, providerClassName matches one of them

Hope this makes it clear

Reply:
yeah. a little confused. Can you also add the right return for each case？

> 1 provider, providerClassName matches the provider <- currently missing

don't know what is missing. The following code should cover this case. is that correct?
```
for (ContextStorageProvider provider : providers) {
      if (provider.getClass().getName().equals(providerClassName)) {
        return provider.get();
      }
    }
```

Reply:
Sure updated my comment, in every other case we log and return default storage.

Yeah starting with that, and than one more branch for if there's a single provider and empty providerClassName seems fine to me.

Reply:
I think the current changes can satisfy all the cases. Just need log in some case. WDYT?

Reply:
Sorry got it! Yeah seems good, really needed to expand the diff